packages/-ember-data/tests/acceptance/relationships/belongs-to-test.js

```diff
@@ -392,6 +392,38 @@ module('async belongs-to rendering tests', function(hooks) {
       assert.equal(this.element.textContent.trim(), 'Kevin has two children and one parent');
     });
 
+    test('Setting an async belongsTo to null, then restoring it', async function(assert) {
+      let people = makePeopleWithRelationshipData();
+      let sedona = store.push({
+        data: people.dict['5:has-parent-no-children'],
+      });
+
+      adapter.setupPayloads(assert, [{ data: people.dict['3:has-2-children-and-parent'] }]);
+
+      // render
+      this.set('sedona', sedona);
+
+      await render(hbs`
+      <p>{{sedona.parent.name}}</p>
+      `);
+
+      assert.equal(this.element.textContent.trim(), 'Kevin has two children and one parent');
+
+      const parent = sedona.get('parent');
```

Comment:
Since this is an `async` relationship, I think the fix for this test is: 

```js
const parent = await sedona.get('parent');
```

Otherwise it is a PromiseProxy.  I think that should fix the test.

Reply:
hmm, if the test is wrong, I need to check my app. I'll get back to you

Reply:
I think the point is here, that it worked in 3.10

If `sedona.parent` is a proxy in the first place, and it works.
it is then set to `null`
it is then set back to the same proxy

..then I would expect it to be the same.

Reply:
I see!  That may make sense if we kept around a local copy of the remove "parent".  However, I'm not sure if that was publicly supported behaviour before 3.10 or not.  Might have to dig around commits in that time frame.